openml-api/src/test/java/com/feedzai/openml/data/schema/DatasetSchemaTest.java

```diff
@@ -116,6 +120,43 @@ public void testValidSchema() {
                 .isNotEqualTo(anotherSchema.hashCode());
     }
 
+    /**
+     * Tests that predictive fields:
+     * <ul>
+     *     <li>Do not include the target field when the schema contains a target field.</li>
+     *     <li>Include all fields when the schema has no field marked as target variable.</li>
+     * </ul>
+     */
+    @Test
+    public final void testPredictiveFieldsDoNotContainTargetField() {
+        final FieldSchema targetField =
+                new FieldSchema("field3", 3, new CategoricalValueSchema(false, ImmutableSet.of("true", "false")));
+
+        final List<FieldSchema> predictiveFields = ImmutableList.of(
+                new FieldSchema("field0", 0, new NumericValueSchema(false)),
+                new FieldSchema("field1", 1, new StringValueSchema(true)),
+                new FieldSchema("field2", 2, new NumericValueSchema(true))
+        );
+
+        final List<FieldSchema> allFields = ImmutableList.<FieldSchema>builder()
+                .addAll(predictiveFields)
+                .add(targetField)
+                .build();
+
+        final DatasetSchema datasetWithTarget = new DatasetSchema(3, allFields);
+        final DatasetSchema datasetNoTarget = new DatasetSchema(allFields);
+
+        assertThat(datasetWithTarget.getPredictiveFields())
```

Comment:
here you don't use soft assertions. what's your reasoning to decide when to use (legit question, I'm not saying you are wrong)?

Reply:
Nicely spotted. I should use! (see my comment above for the explanation on why)